### PR TITLE
feat(metamodel): add CollectionSizeValidator concept for array and map property size constraints

### PR DESCRIPTION
**Pull request by muhabdulkadir**
# Closes #[1292](https://github.com/accordproject/concerto/issues/1292)

Array and map-typed properties currently have implicitly unbounded sizes. This adds the metamodel foundation for enforcing min/max element counts using size=[min,max] syntax.

- sizeValidator lives on abstract Property (not per-subtype) since `isArray` already lives there and any property type can be an array
- Uses the same `[min,max]` bracket pattern as `length=[,]` for strings and `range=[,]` for numbers
- The metamodel namespace remains @1.0.0 — metamodel versioning strategy needs separate discussion with maintainers

### Changes

- Add `CollectionSizeValidator` concept with optional `minSize` and `maxSize` integer fields
- Add optional `sizeValidator` field on `abstract Property` to support size constraints on array properties and map-typed properties

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

**Review by jonathan-casey — APPROVED** [review]

**Comment by jonathan-casey** [comment]
@muhabdulkadir we should add test cases around negative values too?

```
should reject a negative minSize value
should reject a negative maxSize value
```

File: lib/metamodel.cto [hunk 1]
```diff
@@ -143,10 +143,16 @@ abstract concept Property {
   o String name regex=/^(\p{Lu}|\p{Ll}|\p{Lt}|\p{Lm}|\p{Lo}|\p{Nl}|\$|_|\\u[0-9A-Fa-f]{4})(?:\p{Lu}|\p{Ll}|\p{Lt}|\p{Lm}|\p{Lo}|\p{Nl}|\$|_|\\u[0-9A-Fa-f]{4}|\p{Mn}|\p{Mc}|\p{Nd}|\p{Pc}|\u200C|\u200D)*$/u
   o Boolean isArray default=false
   o Boolean isOptional default=false
+  o CollectionSizeValidator sizeValidator optional
   o Decorator[] decorators optional
   o Range location optional
 }
 
+concept CollectionSizeValidator {
+  o Integer minSize optional
```
Comment (jamieshorten) [inline lib/metamodel.cto]: Should there be a range validator on these to ensure they aren't negative?

Comment (muhabdulkadir) [reply lib/metamodel.cto]: This is a runtime validation check, not a metamodel concern.

Comment (jamieshorten) [reply lib/metamodel.cto]: Yep, I guess it follows the pattern of the other validators.
